pychunkedgraph/graph/edits.py

```diff
@@ -96,12 +97,14 @@ def merge_preprocess(
     active_edges = [types.empty_2d]
     inactive_edges = [types.empty_2d]
     for layer in np.unique(edge_layers):
-        layer_edges = subgraph_edges[edge_layers == layer]
-        edges_parents = cg.get_roots(
-            layer_edges.ravel(), time_stamp=parent_ts, stop_layer=layer + 1
-        ).reshape(-1, 2)
-        active_mask = edges_parents[:, 0] == edges_parents[:, 1]
-        active, inactive = layer_edges[active_mask], layer_edges[~active_mask]
+        _edges = subgraph_edges[edge_layers == layer]
+        edge_nodes = fastremap.unique(_edges.ravel())
+        roots = cg.get_roots(edge_nodes, time_stamp=parent_ts, stop_layer=layer + 1)
+        parent_map = dict(zip(edge_nodes, roots))
```

Comment:
I think this can mask issues in the return of the roots. Could you add a check along the lines of
`assert len(roots) == len(edge_nodes)`